docs/working-with-projects/working-with-images/create-tag-immutability-rules.md

```diff
@@ -3,29 +3,29 @@ title: Tag Immutability Rules
+By default, users can repeatedly push an artifact with the same tag to repositories in Harbor. This causes the previous artifact to effectively be overwritten with each push, in that the tag now points to a different artifact and the artifact that previously used the tag now becomes tagless. This is due to the Docker Distribution implementation, that does not enforce the mapping between an image tag and the image digest. This can be undesirable in certain cases, because the tag can no longer be trusted to identify the image version. The sha256 digest remains reliable and always points to the same build, but it is not rendered in a human-readable format.
 
-Moreover, the Docker implementation requires that deleting a tag results in the deletion of all other tags that point to the same digest, causing unwanted image deletions.
-
-To prevent this, Harbor allows you to configure tag immutability at the project level, so that images with certain tags cannot be pushed into Harbor if their tags match existing tags. This prevents existing images from being overwritten. Tag immutability guarantees that an immutable tagged image cannot be deleted, and cannot be altered through repushing, retagging, or replication. 
+To prevent this, Harbor allows you to configure tag immutability at the project level, so that artifacts with certain tags cannot be pushed into Harbor if their tags match existing tags. This prevents existing artifacts from being overwritten. Tag immutability guarantees that an immutable tagged artifact cannot be deleted, and cannot be altered through repushing, retagging, or replication. 
 
 Immutability rules use `OR` logic, so if you set multiple rules and a tag is matched by any of those rules, it is marked as immutable. 
 
 ## How Immutable Tags Prevent Tag Deletion
 
-Tags that share a common digest cannot be deleted even if only a single tag is configured as immutable. For example:
+Tags are that objects attached to artifact, cannot be deleted when it is configured as immutable. You cannot set a immutable artifact as the immutability is affect the tag level only, but if any tag of the artifact is set to immutable, the artifact is immutable. For example:
 
-1. In a project, set an immutable tag rule that matches the image and tag `hello-world:v1`.
-1. In the Docker client, pull `hello-world:v1` and retag it to `hello-world:v2`.
+1. In the Docker client, push `hello-world:v1` into a project.
+1. In the same project, set an immutable tag rule that matches the image and tag `hello-world:v1`.
```

Comment:
-> ' set an immutable tag rule in this project that matches ..'

Reply:
updated